### PR TITLE
Site Creation: Remove feature flag and legacy code.

### LINKED ISSUE #10329: [Site creation] Remove feature flag & legacy site creation code
In #10312, we introduced the feature flag for site creation. In #10341, we'll activate it for all build configurations in anticipation of launching the feature.

Post-release, we'll make a determination that site creation is successful, and it will emerge as the sole experience. At that time, we can remove the feature flag altogether. We can also remove the existing site creation code.

Project master thread: `p9wMUP-bk-p2`

Plan proposal: `p9wMUP-bs-p2`

### PR DESCRIPTION
**Pull request by diegoreymendez**
### Description:

Fixes #10329 

This PR removes the legacy Site Creation code and the feature flag.

This PR does not make any changes to any old tracking code for the old Site Creation.  We'll address that separately to keep the scope of this PR atomic.

### Testing:

Make sure the App builds.
Make sure the unit tests run.
Make sure Site Creation still works fine.

### CODE REVIEW

File: WordPress/WordPress.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1733,7 +1718,7 @@
 		FF00889F204E01AE007CCE66 /* MediaQuotaCell.swift in Sources */ = {isa = PBXBuildFile; fileRef = FF00889E204E01AE007CCE66 /* MediaQuotaCell.swift */; };
 		FF0148E51DFABBC9001AD265 /* NSFileManager+FolderSize.swift in Sources */ = {isa = PBXBuildFile; fileRef = FF0148E41DFABBC9001AD265 /* NSFileManager+FolderSize.swift */; };
 		FF0AAE0A1A150A560089841D /* WPProgressTableViewCell.m in Sources */ = {isa = PBXBuildFile; fileRef = FF0AAE091A150A560089841D /* WPProgressTableViewCell.m */; };
-		FF0AAE0D1A16550D0089841D /* BuildFile in Sources */ = {isa = PBXBuildFile; };
+		FF0AAE0D1A16550D0089841D /* (null) in Sources */ = {isa = PBXBuildFile; };
```
Comment (shiki) [inline WordPress/WordPress.xcodeproj/project.pbxproj]: 🤔

File: WordPress/Classes/ViewRelated/NUX/SiteCreationCreateSiteViewController.swift [hunk 2]
```diff
@@ -1,350 +0,0 @@
...
-        WPStyleGuide.configureColors(for: view, andTableView: nil)
-        addWordPressLogoToNavController()
-        // Remove help button.
-        navigationItem.rightBarButtonItems = nil
-        // Remove Back button. There's no going back now!
-        navigationItem.hidesBackButton = true
-    }
-
-    func setLabelText() {
-        layingFoundationLabel.text = NSLocalizedString("Laying site foundation...", comment: "Text shown during the site creation process when it is on the first step.")
```
Comment (shiki) [inline WordPress/Classes/ViewRelated/NUX/SiteCreationCreateSiteViewController.swift]: The removed strings should be automatically removed from `Localizable.strings`, right?

Comment (diegoreymendez) [reply WordPress/Classes/ViewRelated/NUX/SiteCreationCreateSiteViewController.swift]: They should be automatically removed when the strings are updated through our scripts, yes.  It's not a manual process.

File: WordPress/WordPress.xcodeproj/project.pbxproj (hunk 1 above)
Comment (diegoreymendez) [reply WordPress/WordPress.xcodeproj/project.pbxproj]: Removed the line that was being referenced.

**Pull request merged**